test/index.js

```diff
@@ -61,14 +61,14 @@ test('base', t => {
     ['path/**/*', 'path'],
     ['path/**/subdir/foo.*', 'path'],
     ['path/subdir/**/foo.js', 'path/subdir'],
-    ['path/!subdir/foo.js', 'path/!subdir'],
+    ['path/!subdir/foo.js', 'path'],
```

Comment:
The `!subdir`  should 100% be treated as a pattern imo, leaving `path` as the only static portion.